src/extract/path/de.rs

```diff
@@ -561,18 +560,16 @@ mod tests {
         a: i32,
     }
 
-    fn create_url_params<I, K, V>(values: I) -> UrlParams
+    fn create_url_params<I, K, V>(values: I) -> Vec<(ByteStr, PercentDecodedByteStr)>
     where
         I: IntoIterator<Item = (K, V)>,
         K: AsRef<str>,
         V: AsRef<str>,
     {
-        UrlParams(
-            values
-                .into_iter()
-                .map(|(k, v)| (ByteStr::new(k), ByteStr::new(v)))
-                .collect(),
-        )
+        values
+            .into_iter()
+            .map(|(k, v)| (ByteStr::new(k), PercentDecodedByteStr::new(v).unwrap()))
```

Comment:
I think this `.unwrap()` should be an `.expect("message")`. Without having read the rest of this PR I can't tell why this wouldn't sometimes cause a panic.

Reply:
This is test code.